### ISSUE TITLE
Date.toIsoString returning unexpected negative result

### ISSUE THREAD

**Issue by mordrax** [justinmimbs/elm-date-basic#6]
```
> Date.fromCalendarDate 0 Date.Jan 1
RD -364 : Date.Date

> Date.fromCalendarDate 0 Date.Jan 1 |> Date.toIsoString
"0001-01--364" : String

-- it should be as per the input day
"0001-01-1" : String
```

**Comment by justinmimbs** [comment]
Ah! Thank you. I haven't taken into consideration values before `"0001-01-01"`. Yes, it should produce:

```elm
Date.fromCalendarDate 0 Date.Jan 1 |> Date.toIsoString
-- "0000-01-01"
```

I'll get it fixed. :thumbsup:

**Comment by mordrax** [comment]
I think my backend guy just gets a kick out of setting ridiculous min dates. I told him he couldn't have a 0 for month... :/

**Comment by justinmimbs** [comment]
Nice! This is fixed in [2.0.2](https://package.elm-lang.org/packages/justinmimbs/date/2.0.2/). And now your backend guy can even send dates with negative years if he'd like. :smile:

Thanks for the report!

```elm
import Date
import Time exposing (Month(..))

Date.fromCalendarDate 0 Jan 1
-- RD -365

Date.fromCalendarDate 0 Jan 1 |> Date.toIsoString
-- "0000-01-01"

Date.fromCalendarDate 0 Jan 1 |> Date.toIsoString |> Date.fromIsoString
-- Ok (RD -365)
```
